processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -274,7 +274,9 @@ else if (componentTypeMirror.getKind().isPrimitive()) {
         else {
             isEnumType = false;
             isInterface = false;
-            name = mirror.toString();
+            // When the component type is primitive and is annotated with ElementType.TYPE_USE then
+            // the typeMirror#toString returns (@CustomAnnotation :: byte) for the javac compiler
+            name = mirror.getKind().isPrimitive() ? NativeTypes.getName( mirror.getKind() ) : mirror.toString();
```

Comment:
I'm not in favor of getting the name via the `mirror.toString()` at all. Can we go via the element somehow? I see somewhat up is the same construction.. (line 263).. 

If not.. go ahead and merge.. I think I even investigated this before in the exact same spot 😄

Reply:
We do that when the type is a declared type. Otherwise there is no element, there is only a `TypeKind` and we do use it then the type is primitive. Not sure if we can do more here. 

>I think I even investigated this before in the exact same spot 😄

We did this together for #1558